### ISSUE TITLE
[Feature Request]: Right side bar status idea

### ISSUE THREAD

**Issue by kadisonm** [sboesen/remotely-secure#43]
### What feature are you suggesting?

Hi, I wanted to move this here from #41.

I think personally the right side bar would be the best option for viewing sync status purely because there isn't really any other option with Obsidian at the moment (that I'm aware about) 

But I do understand that if you display the sync status here there is a lot of pointless space not being used in the sidebar. But maybe some other things could go there, or could be valuable for in the future. Maybe something like the files that have been changed and are yet to be synced (if sync on save isn't on)

I would like to discuss more about it because it is an option for displaying sync status, and maybe there are more options too?

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by kadisonm** [comment]
That and if people regularly use other views like backlinks and calendar then it wouldn’t really be of use to them.

In my case it works since I just have to swipe right, check it synced then close the app. But I understand that’s a very niche case. 

Preferably it would be good if it could be displayed with the word count so it shows on any view. But I have a feeling it’s not possible.

**Comment by sboesen** [comment]
> Would it be possible to add the sync status to the right hand menu on mobile? This is where you find the word count and backlinks. I asked on the Obsidian discord, so I'll let you all know once I get a response.

Did you get a response? I haven't looked into it but wondering if it's even possible given the Obsidian APIs to add something in the sidebar. If it's possible I'm not strongly opposed to adding the status in the sidebar as an option disabled by default - it wouldn't add too much code complexity. 

I also want to take time and think if there's a different way to display this on mobile that makes sense.

**Comment by kadisonm** [comment]
> Did you get a response?

Oh yep. Some plugins like Calendar do. From what I've gathered but haven't tested. You get the right leaf and set it's view to your own. It would be preferable to add the status under the word count instead  but I don't think it's possible, I'll ask on the discord.

https://github.com/sboesen/remotely-secure/assets/134670047/81ca5ef0-7407-49d5-ab29-2d1da02e629b

In terms of the view, showing the files queued for syncing could also be accompanied with a sync button. Something like this to make it worth using the side bar on desktop or mobile.

![Group 6](https://github.com/sboesen/remotely-secure/assets/134670047/bde9a359-a281-4326-8dec-98dca54810ab)

I'll also look for some other options for the sync status.

**Comment by kadisonm** [comment]
I just realised this probably wouldn’t work as file changes are only checked with sync on save to then call a sync. I assume changes aren’t stored. Maybe alternatively it could be a sync history?

**Comment by kadisonm** [comment]
I'd imagine it would look like this:

![image](https://github.com/sboesen/remotely-secure/assets/134670047/c281bbf9-2563-4d35-92b7-e6578f3a8d0b)

To solve the last minute problem a toggle in the settings to display seconds would work, it does look quite cluttered so I think it would default to off. 
Only thing is sync on save runs quite a lot so it would get cluttered with a lot of pages. So maybe an option to group syncs within the past x minutes (could be a setting) would work:

![Group8](https://github.com/sboesen/remotely-secure/assets/134670047/d154284a-34ad-43d7-9056-8022c1b4f471)

This isn't inherently a solution to the sync status problem. But it makes having a right bar make sense and pretty much serves as a sync status anyways. Let me know what you think.

**Comment by kadisonm** [comment]
Also this would be a decent amount of code and CSS, so the design could be flattened a lot. I think it's visually appealing though.

**Comment by sboesen** [comment]
I agree, definitely looks great! I don't think it looks too cluttered. Appreciate you taking the time to do a mockup of what you were describing.

I am pretty bad at frontend design so that would be some work for me. If you (or anyone else) have any interest in that part let me know.

**Comment by kadisonm** [comment]
Hi, I can definitely make a pull request and try to make this once I'm done with school for this year. But that would be in a week or two.

I'm aware you probably have a lot of other pressing things to focus on. So I can try to do all the coding as well if you would like.

**Comment by sboesen** [comment]
That would be great! Definitely appreciate you taking a stab at it. I like the idea & seeing the picture definitely helps drive home the value of the feature for mobile.

**Comment by MartinJDavis93** [comment]
Originally, I thought of writing a long comment but decided not to waste time. The only request, if you decide to implement it, please add the option to disable it in the plugin settings. I'd greatly appreciate it.

**Comment by kadisonm** [comment]
> The only request, if you decide to implement it, please add the option to disable it in the plugin settings. 

Of course, it would probably be best to as we'd have to store the sync history (probably in the json file?), and there is no point storing it if the user doesn't use the history. 

Please let me know if you have any other thoughts or concerns though, because I'd like to try to make this as good as possible. 
:)

**Comment by kadisonm** [comment]
Hi, I had a look through the source code and had a look into the localisation. My understanding of it is that the i18n class is made onLoad and the language is passed into the object. This makes sense and I assume the i18n.t() takes in the english string and returns the translated string. I just have a few questions since I'll probably run into this when I start.

What does i18n stand for? 😆
Why is there a t function declared multiple times in main.ts as opposed to using this.i18n.t()?
And lastly, if I add an english item to the lang repository how does this affect the other languages? Do I just add a place holder for someone else to translate?

**Comment by sboesen** [comment]
Hehe yeah, it's useful. We have three translation files currently, so each will need to be updated. 

> Why is there a t function declared multiple times in main.ts as opposed to using this.i18n.t()?

Copy/paste instead of reusing. Please feel free to refactor this! Lots of code duplication which I'm cleaning up slowly when I touch the related areas.

The sync code is in a half-mess state because of some wrapper functions I added, but it's usable and a stepping stone to improving sync on save to handle .obsidian file changes, too.

**Comment by kadisonm** [comment]
I will start on this soon as I'm on holidays now. Just have a couple of my own things and some school work to finish up.

**Comment by kadisonm** [comment]
On the topic of sync status. What about a context menu item that makes a notice with the sync status. This could also be a command that could be added to the ribbon.

I already have a context menu button to run sync but I find if I'm about to close the app I just want to check if sync on save ran, but in most cases it will start a new sync and I'll have to wait.

![Screenshot_20231228_035249](https://github.com/sboesen/remotely-sync/assets/134670047/5f046c84-12a4-4056-8e42-300a92fcd47b)

Thoughts? I still think sync history is a handy feature and I still plan on making it, but this could also help the issue.

**Comment by sboesen** [comment]
Hmm, not opposed to adding this but it still feels clunky. I think having options is good and people can pick what works best since there is no obvious fix for mobile which everyone likes (sync status bar is more popular on desktop for sure, but even then some of our users prefer the notification popups).

**Comment by kadisonm** [comment]
I'm going to get started on this now but we need to sort out storing the sync data.

It might be best to store and read the data from the remote so the history will update if another device syncs. Might need a way to identify if the sync was external or local. Maybe giving each device an id?

This could also tell the user that they're not fully synced. And maybe prompt them to sync if the last sync was external. I think that's another feature we discussed before.

**Comment by kadisonm** [comment]
Hi, I tried uploading a sync history to the metadata in a fork and got it working. Not sure how to share the commit so I'll add a [link](https://github.com/kadisonm/remotely-sync/commit/319fbce5aa2e2dce9b6119cd5a74ae5e1e8616ec) please have a look.

Let me know if there is another way to store the history, or anything you want me to change with how I am uploading it to the metadata. :)

**Comment by sboesen** [comment]
> This could also tell the user that they're not fully synced. 

I think we would need to implement https://github.com/sboesen/remotely-sync/issues/47 to get this to work. Otherwise how would it know it hasn't synced? Agreed this would be useful, though (and help solve #47!)

> Not sure how to share the commit so I'll add a [link](https://github.com/kadisonm/remotely-sync/commit/319fbce5aa2e2dce9b6119cd5a74ae5e1e8616ec) please have a look. (and any comments I added in particular)

Please open a PR! If you go to the [main repo page ](https://github.com/kadisonm/remotely-sync/) there should be a button to open one. That is the best place to review and discuss code changes, can add comments inline etc. Overall looks good, though, really appreciate you taking the time to implement this feature! Looks like it should help close at least two issues.

**Comment by kadisonm** [comment]
> I think we would need to implement https://github.com/sboesen/remotely-sync/issues/47 to get this to work. Otherwise how would it know it hasn't synced? Agreed this would be useful, though (and help solve https://github.com/sboesen/remotely-sync/issues/47!)

100% I agree. Would probably be best to solve that issue first.

**Comment by lyiton** [comment]
I have updated the plugin to 0.4.36, but I cannot see the sync status in the right column, and there is no "Get sync status" option in the plugin's settings. What is the problem?
![Screenshot_2024-01-08-13-51-01-700_md obsidian](https://github.com/sboesen/remotely-sync/assets/76651858/f54d60a9-68a5-430e-a5d1-0aa23734ab2c)

**Comment by kadisonm** [comment]
Hi. The get sync status is a command on mobile.

You can either use the command or you can access it by going to settings > mobile

**Comment by lyiton** [comment]
Aha, I'm sorry, I didn't pay attention, thank you!!

**Comment by kadisonm** [comment]
No worries. I still plan to add the sync history but added this to provide different options for the getting the status.

**Comment by sboesen** [comment]
Yeah worth adding documentation for all of the supported plugin commands.

Did not know about mobile quick action!

**Comment by kadisonm** [comment]
> Did not know about mobile quick action!

I just realised today as well. I think I'd rather have the sync history and use the quick action as quick switcher. But for now it's pretty good for me.
